### PR TITLE
Implement the structure for adding API and hand-written documentation

### PR DESCRIPTION
**Pull request by dsnopek**
This PR attempts to setup the structure to allow us to add documentation!

For in-editor API documentation, this PR depends on:

- Godot PR https://github.com/godotengine/godot/pull/91518
  -  This allows using `godot --doctool  --gdextension-docs` to update XML documentation for GDExtensions
- godot-cpp PR https://github.com/godotengine/godot-cpp/pull/1374
  - This  allows godot-cpp to send those XML docs to the Godot editor when loaded

However, this also includes a Sphinx setup based on [godot-docs](https://github.com/godotengine/godot-docs), which can convert that XML into RST for online API documentation, as well as allowing us to add hand-written RST for tutorials and other more wordy explanations.

Here's some screenshots of the HTML docs:

Selection_154

Selection_153

There are GitHub actions included that will automatically build these docs and push them to GitHub pages, when changes are pushed to the `master` branch.

You can see the [GitHub pages on my fork](https://dsnopek.github.io/godot_openxr_vendors/index.html), which I used to test that the GitHub actions were working.

This PR doesn't include much in the way of actual documentation, it's meant to setup the structure, and then we can actually write some docs in follow-up PRs. However, for testing, I added API documentation for [OpenXRFbRenderModel](https://dsnopek.github.io/godot_openxr_vendors/api/class_openxrfbrendermodel.html) (link to GitHub pages on my fork).

The `docs/README.md` explains how you can build the docs locally.

_Note: It appears we missed the argument names when binding some methods, so I also added those in this PR, so that it would stop complaining about them being missing, when generating the docs :-)_

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+sys.path.append(os.path.abspath('_extensions'))
+
+# -- Project information -----------------------------------------------------
+# https://www.sphinx-doc.org/en/master/usage/configuration.html#project-information
+
+project = 'Godot OpenXR Vendors plugin'
+copyright = (
+    "2014-present Juan Linietsky, Ariel Manzur and the Godot community (CC BY 3.0)"
+)
+author = "Juan Linietsky, Ariel Manzur and the Godot community"
```
Comment (m4gr3d) [inline docs/conf.py:17]: The copyright and author should be updated to match the ones used for this repo.

**Review by m4gr3d — APPROVED** [review]

**Comment by m4gr3d** [comment]
Looks great!

File: doc_classes/KhronosEditorExportPlugin.xml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<class name="KhronosEditorExportPlugin" inherits="OpenXREditorExportPlugin" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="../class.xsd">
```
Comment (m4gr3d) [inline doc_classes/KhronosEditorExportPlugin.xml:2]: @akien-mga left a comment on the godot-cpp PR about the invalid link to `class.xsd`.

**Comment by dsnopek** [comment]
Looks like the GitHub actions worked! The docs (such as they are) have been pushed to GitHub pages:

https://godotvr.github.io/godot_openxr_vendors/

**Pull request merged**